samples/dashboards/demo/mqtt/demo.js

```diff
@@ -0,0 +1,1520 @@
+                y: -80
+            },
+            labels: {
+                distance: '100%',
+                y: 5,
+                align: 'auto'
+            },
+            minorTicks: false,
+            tickAmount: 1,
+            visible: true,
+            min: 0,
+            max: 0 // Populated on update
+        },
+        series: [{
+            name: measInfo.brief('P_gen'),
+            enableMouseTracking: true,
+            innerRadius: '80%',
+            radius: '120%'
+        }],
+        tooltip: {
+            valueSuffix: ' ' + powerUnit
+        }
+    }
+};
+
+// Chart for displaying generated power.
+const chartComponent = {
+    type: 'Highcharts',
+    renderTo: 'el-chart',
+    connector: {
+        id: defaultConnId,
+        columnAssignment: [{
+            seriesId: measInfo.brief('P_gen'),
+            data: ['time', 'power']
+        }]
+    },
+    sync: {
+        highlight: {
+            enabled: true,
+            autoScroll: true
+        }
+    },
+    chartOptions: {
+        ...commonChartOptions,
+        chart: {
+            type: 'spline',
+            animation: true
+        },
+        credits: {
+            enabled: false
+        },
+        xAxis: {
+            type: 'datetime'
+        },
+        yAxis: {
+            min: 0,
+            max: 0, // Populated on update
+            title: {
+                text: measInfo.descr('P_gen')
+            }
+        },
+        tooltip: {
+            valueSuffix: ' ' + powerUnit
+        }
+    }
+};
+
+// Datagrid displaying the history of generated power
+// over the last 'n' hours. Oldest measurements at the top.
+const datagridComponent = {
+    type: 'DataGrid',
+    renderTo: 'el-datagrid',
+    connector: {
+        id: defaultConnId
+    },
+    sync: {
+        highlight: {
+            enabled: true,
+            autoScroll: true
+        }
+    },
+    dataGridOptions: {
+        credits: {
+            enabled: false
+        },
+        columns: [{
+            id: 'time',
+            header: {
+                format: 'Time (local)'
+            },
+            cells: {
+                formatter: function () {
+                    return Highcharts.dateFormat(
+                        '%H:%M:%S',
+                        this.value
+                    );
+                }
+            }
+        }, {
+            id: 'power',
+            header: {
+                format: measInfo.descr('P_gen')
+            },
+            cells: {
+                format: '{value:.2f}'
+            }
+        }]
+    }
+};
+
+// Print log message
+function printLog(msg) {
+    if (log) {
+        log(msg);
+    }
+}
+
+// Set Dashboard visibility. The control bar is always visible.
+function setVisibility(visible) {
+    const cells = document.getElementsByClassName('row');
+
+    for (let i = 1; i < cells.length; i++) {
+        const el = cells[i];
+        el.style.display = visible ? 'flex' : 'none';
+    }
+}
```

Comment:
It would be better practice to toggle a class name in the top container (one line of JS), then use CSS to apply cell visibility.